wellregistry/registry/admin/monitoring_location.py

```diff
@@ -217,10 +220,100 @@ def _has_permission(perm, user, obj=None):
     return user.has_perm(perm) and (not obj or obj.agency.agency_cd in _get_groups(user))
 
 
+def _get_usergroups(request):
+    user_groups = ""
+    if not request.user.is_superuser:
+        user_groups = re.search(
+            "\[(.*)\]", str(_get_groups(request.user))).group(1)
+        user_groups = " where agency.agency_cd IN (" + user_groups + ") "
+    return user_groups
+
+
+def _get_where_clause(request, user_groups):
+    where_clause = ""
+    sep = ""
+    query_params = request.body.decode().split("&")
+    for query_param in query_params:
+        param = query_param.split("=")
+        if param[0] == "_selected_action":
+            where_clause += sep + param[1]
+            sep = ","
+    if len(where_clause) > 0:
+        if user_groups == "":
+            where_clause = " where registry_monitoringlocation.id IN (" + \
+                where_clause + ") \n"
+        else:
+            where_clause = user_groups + " and registry_monitoringlocation.id IN (" + \
+                where_clause + ") \n"
+    return where_clause
+
+
+def _get_monitoringlocation_queryset(request):
+    user_groups = _get_usergroups(request)
+    where_clause = _get_where_clause(request, user_groups)
+    query_ml = "SELECT agency.agency_cd, registry_monitoringlocation.site_no as site_no, \n\
```

Comment:
since you have all the other fields on their own line, it might be nice to have site_no on its own line too

Reply:
fixed